xrootd/cmd/xrd-srv/main_test.go

```diff
@@ -0,0 +1,194 @@
+		errorHandler(errors.Wrap(err, "xrd-srv: an error occured"))
+	})
+
+	go func() {
+		if err = srv.Serve(listener); err != nil && err != server.ErrServerClosed {
+			errorHandler(errors.Wrap(err, "xrd-srv: could not serve"))
+		}
+	}()
+
+	return srv, baseDir, nil
+}
+
+func createClient() (*client.Client, error) {
+	return client.NewClient(context.Background(), addr, "gopher")
+}
+
+func TestHandler_Dirlist(t *testing.T) {
+	srv, baseDir, err := createServer(func(err error) {
+		t.Error(err)
+	})
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer srv.Shutdown(context.Background())
+	defer os.RemoveAll(baseDir)
+
+	file := path.Join(baseDir, "file1.txt")
+	f, err := os.Create(file)
+	if err != nil {
+		t.Fatalf("could not create test file: %v", err)
+	}
+	f.Close()
+
+	fileInfo, err := os.Stat(file)
+	if err != nil {
+		t.Fatalf("could not stat test file: %v", err)
+	}
+
+	cli, err := createClient()
+	if err != nil {
+		t.Fatalf("could not create client: %v", err)
+	}
+	defer cli.Close()
+
+	got, err := cli.FS().Dirlist(context.Background(), "/")
+	if err != nil {
+		t.Fatalf("could not call Dirlist: %v", err)
+	}
+
+	want := []xrdfs.EntryStat{
+		{
+			EntryName:   "file1.txt",
+			HasStatInfo: true,
+			Flags:       xrdfs.StatIsWritable | xrdfs.StatIsReadable,
+			Mtime:       fileInfo.ModTime().Unix(),
+		},
+	}
+
+	if !reflect.DeepEqual(got, want) {
+		t.Fatalf("wrong Dirlist response:\ngot = %v\nwant = %v", got, want)
+	}
+}
+func TestHandler_Dirlist_WhenPathIsInvalid(t *testing.T) {
+	srv, baseDir, err := createServer(func(err error) {
+		t.Error(err)
+	})
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer srv.Shutdown(context.Background())
+	defer os.RemoveAll(baseDir)
+
+	cli, err := createClient()
+	if err != nil {
+		t.Fatalf("could not create client: %v", err)
+	}
+	defer cli.Close()
+
+	_, err = cli.FS().Dirlist(context.Background(), "/path/not/exist")
+	serverError, ok := err.(xrdproto.ServerError)
+	if !ok {
+		t.Fatalf("could not cast err to ServerError: %v", err)
+	}
+	if serverError.Code != xrdproto.IOErrorCode {
+		t.Fatalf("wrong error code:\ngot = %v\nwant = %v", serverError.Code, xrdproto.IOErrorCode)
+	}
+}
+
+func TestHandler_Dirlist_With1000Requests(t *testing.T) {
+	srv, baseDir, err := createServer(func(err error) {
+		t.Error(err)
+	})
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer srv.Shutdown(context.Background())
+	defer os.RemoveAll(baseDir)
+
+	file := path.Join(baseDir, "file1.txt")
+	f, err := os.Create(file)
+	if err != nil {
+		t.Fatalf("could not create test file: %v", err)
+	}
+	f.Close()
+
+	cli, err := createClient()
+	if err != nil {
+		t.Fatalf("could not create client: %v", err)
+	}
+	defer cli.Close()
+
+	var wg sync.WaitGroup
+	wg.Add(1000)
+	for i := 0; i < 1000; i++ {
+		go func() {
+			defer wg.Done()
+			_, err := cli.FS().Dirlist(context.Background(), "/")
+			if err != nil {
+				t.Fatalf("could not call Dirlist: %v", err)
+			}
+		}()
+	}
+	wg.Wait()
+}
+
+func BenchmarkHandler_Dirlist(b *testing.B) {
```

Comment:
could you add the benchmark numbers in the commit message?
(to give a point of reference.)

Reply:
FYI, I get:

```
$> go test -bench Handler -benchmem -run=NONE -count 20  |& tee ref.txt
goos: linux
goarch: amd64
pkg: go-hep.org/x/hep/xrootd/cmd/xrd-srv
BenchmarkHandler_Dirlist-8   	   30000	     42425 ns/op	   11889 B/op	      50 allocs/op
BenchmarkHandler_Dirlist-8   	   30000	     42657 ns/op	   11889 B/op	      50 allocs/op
BenchmarkHandler_Dirlist-8   	   30000	     42393 ns/op	   11889 B/op	      50 allocs/op
BenchmarkHandler_Dirlist-8   	   30000	     42801 ns/op	   11889 B/op	      50 allocs/op
BenchmarkHandler_Dirlist-8   	   30000	     42710 ns/op	   11889 B/op	      50 allocs/op
BenchmarkHandler_Dirlist-8   	   30000	     42453 ns/op	   11889 B/op	      50 allocs/op
BenchmarkHandler_Dirlist-8   	   30000	     42533 ns/op	   11889 B/op	      50 allocs/op
BenchmarkHandler_Dirlist-8   	   30000	     42576 ns/op	   11889 B/op	      50 allocs/op
BenchmarkHandler_Dirlist-8   	   30000	     42674 ns/op	   11889 B/op	      50 allocs/op
BenchmarkHandler_Dirlist-8   	   30000	     42401 ns/op	   11889 B/op	      50 allocs/op
BenchmarkHandler_Dirlist-8   	   30000	     42386 ns/op	   11889 B/op	      50 allocs/op
BenchmarkHandler_Dirlist-8   	   30000	     42601 ns/op	   11889 B/op	      50 allocs/op
BenchmarkHandler_Dirlist-8   	   30000	     42427 ns/op	   11889 B/op	      50 allocs/op
BenchmarkHandler_Dirlist-8   	   30000	     42673 ns/op	   11889 B/op	      50 allocs/op
BenchmarkHandler_Dirlist-8   	   30000	     42585 ns/op	   11889 B/op	      50 allocs/op
BenchmarkHandler_Dirlist-8   	   30000	     42509 ns/op	   11889 B/op	      50 allocs/op
BenchmarkHandler_Dirlist-8   	   30000	     42839 ns/op	   11889 B/op	      50 allocs/op
BenchmarkHandler_Dirlist-8   	   30000	     42781 ns/op	   11889 B/op	      50 allocs/op
BenchmarkHandler_Dirlist-8   	   30000	     42711 ns/op	   11889 B/op	      50 allocs/op
BenchmarkHandler_Dirlist-8   	   30000	     42796 ns/op	   11889 B/op	      50 allocs/op
PASS
ok  	go-hep.org/x/hep/xrootd/cmd/xrd-srv	34.205s

$> benchstat ./ref.txt 
name               time/op
Handler_Dirlist-8  42.6µs ± 1%

name               alloc/op
Handler_Dirlist-8  11.9kB ± 0%

name               allocs/op
Handler_Dirlist-8    50.0 ± 0%
```

Reply:
yep. (I posted it before seeing you had modified the commit message)

for the record, here is my hardware:
```
$> df /
Filesystem      Size  Used Avail Use% Mounted on
/dev/nvme0n1p3  463G  120G  321G  28% /
```
(ie: a SSD disk)